File: app/model/feedback/Mark.java
package model.feedback;

public enum Mark {
  
  // @formatter:off
  VERY_GOOD     (1),
  GOOD          (2),
  NEUTRAL       (3),
  BAD           (4),
  VERY_BAD      (5),
  
  NO_MARK       (-1);
  // @formatter:on
  
  private int value;
  
  private Mark(int theMark) {
    value = theMark;
  }
  
  public int getMark() {
    return value;
  }
  
  public String display(EvaluatedAspect evaledAspect) {
    switch(this) {
    case VERY_GOOD:
      return "Sehr " + evaledAspect.getPositive().toLowerCase();
    case GOOD:
      return evaledAspect.getPositive();
    case NEUTRAL:
      return evaledAspect.getNeutral();
    case BAD:
      return evaledAspect.getNegative();
    case VERY_BAD:
      return "Sehr " + evaledAspect.getNegative().toLowerCase();
    case NO_MARK:
    default:
      return "Keine Angabe";
    }
  }
  
}

File: app/model/feedback/Mark.scala
package model.feedback

import enumeratum.{EnumEntry, PlayEnum}

sealed abstract class Mark(val value: Int, val display: EvaluatedAspect => String) extends EnumEntry

object Mark extends PlayEnum[Mark] {

  override val values: IndexedSeq[Mark] = findValues

  case object VeryGood extends Mark(1, ea => s"Sehr ${ea.positive.toLowerCase}")

  case object Good extends Mark(2, _.positive)

  case object Neutral extends Mark(3, _.neutral)

  case object Bad extends Mark(4, _.negative)

  case object VeryBad extends Mark(5, ea => s"Sehr ${ea.negative.toLowerCase}")

  case object NoMark extends Mark(-1, _ => "Keine Angabe")

}
